### PR TITLE
Ensure that artifact and partition images' checksums are equal.

### PR DESCRIPTION
**Pull request by kacf**
We also switch the data partition to use the same mechanism, mostly as
a free testing mechanism for our dataimg image type.

A test is added for ubimg and the corresponding part image test is in
the mender-image-tests repository.

MEN-2597

Changelog: Title

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>

### CODE REVIEW

**Comment by kacf** [comment]
Goes together with https://github.com/mendersoftware/mender-image-tests/pull/7

**Comment by kacf** [comment]
Rerunning after [x86](https://s3-eu-west-1.amazonaws.com/mender-testing-reports/integration-reports-qemux86-64-uefi-grub/3701/report.html) and [arm](https://s3-eu-west-1.amazonaws.com/mender-testing-reports/integration-reports-vexpress-qemu/3701/report.html) failed.

File: meta-mender-core/classes/mender-part-images.bbclass [hunk 1]
```diff
@@ -42,19 +42,6 @@ IMAGE_NAME_SUFFIX = ""
...
-    existing_exclude_paths = ""
-    for path in exclude_path.split():
-        if os.path.lexists(os.path.realpath(os.path.join(rootfs, path))):
-           existing_exclude_paths += path + " "
-
-    if existing_exclude_paths:
-        return "--exclude-path %s" % existing_exclude_paths
-    else:
-        return ""
-
```
Comment (mirzak) [inline meta-mender-core/classes/mender-part-images.bbclass]: Does this mean that we are no longer excluding `/boot` and `/data` from the rootfs?

Comment (kacf) [reply meta-mender-core/classes/mender-part-images.bbclass]: `rawcopy` uses the `ext4` file, where they should already be removed. The difference is that `wic` is not doing that job anymore.

Comment (mirzak) [reply meta-mender-core/classes/mender-part-images.bbclass]: That is even better :)

**Comment by kacf** [comment]
Thanks!

**Pull request merged**